app/controllers/assignments_controller.rb

```diff
@@ -655,14 +655,12 @@ def get_file_info(file_name, file, assignment_id, path)
   def process_assignment_form(assignment)
     num_files_before = assignment.assignment_files.length
     assignment.assign_attributes(assignment_params)
-    assignment.repository_folder = assignment_params[:short_identifier]
-    new_required_files = assignment.only_required_files_changed? ||
-                         assignment.is_hidden_changed? ||
-                         assignment.assignment_files.any? { |file| file.changed? }
+    assignment.repository_folder = assignment_params[:short_identifier] unless assignment.is_peer_review?
     assignment.save!
-    new_required_files = new_required_files ||
+    new_required_files = assignment.saved_change_to_only_required_files? ||
+                         assignment.saved_change_to_is_hidden? ||
+                         assignment.assignment_files.any?(&:changed?) ||
```

Comment:
Doesn't `changed?` need to be renamed as well?

Reply:
good catch, I thought I had updated that one as well.